test/solidity/transaction/transaction.js

```diff
@@ -0,0 +1,118 @@
+const {
+  Hbar,
+} = require("@hashgraph/sdk");
+
+describe('PrngSystemContract tests', function () {
+  let contractTr, contractSec, wallet, senderWalletAddr
+
+  before(async function () {
+    const factoryTrasnactionSol = await ethers.getContractFactory(
+      Constants.Contract.Transaction
+    )
+    const factorySecondarySol = await ethers.getContractFactory(
+      Constants.Contract.Secondary
+    )
+
+    contractSec = await factorySecondarySol.deploy()
+    await contractSec.deployed();
+    contractTr = await factoryTrasnactionSol.deploy(contractSec.address)
+    await contractTr.deployed();
+
+    const signers = await ethers.getSigners()
+    wallet = signers[0];
+    senderWalletAddr = await wallet.getAddress()
+  })
+
+  it('gasleft() returns (uint256): remaining gas', async function () {
+    const gasLeft = await contractTr.checkGasleft()
+    
+    expect(gasLeft).to.exist
+    expect(ethers.BigNumber.isBigNumber(gasLeft)).to.be.true
+  })
+
+  it('msg.data (bytes calldata): complete calldata', async function () {
+    const myString = 'Hello, world!';
+    const txRes = await contractTr.getMessageData(12, myString)
+    const returnedData = txRes.data
+
+    const ABI = [
+        "function getMessageData(uint integer, string memory inputMessage)"
+    ];
+    const interface = new ethers.utils.Interface(ABI);
+    const encodedFunction = interface.encodeFunctionData("getMessageData", [ 12, myString ])
+    
+    expect(returnedData).to.exist
+    expect(returnedData).to.be.equal(encodedFunction);
+  })
+
+  it('msg.sender (address): sender of the message (current call)', async function () {
+    const sender = await contractTr.getMessageSender()
+
+    expect(sender).to.exist
+    expect(sender).to.be.equal(senderWalletAddr);
+  })
+
+  it('msg.sig (bytes4): first four bytes of the calldata (i.e. function identifier)', async function () {
+    const msgSig = await contractTr.getMessageSignature()
+
+    const ABI = [
+        "function getMessageSignature()"
+    ];
+    const interface = new ethers.utils.Interface(ABI);
+    const encodedFunctionSig = interface.encodeFunctionData("getMessageSignature")
+
+    expect(msgSig).to.exist
+    expect(msgSig).to.be.equal(encodedFunctionSig);
+  })
+
+  it('msg.value (uint): number of wei sent with the message', async function () {
+    const valueToSend = ethers.utils.parseEther(String(1));
+    const txRes = await contractTr.getMessageValue({value: valueToSend});
+    const receipt = await txRes.wait()
+    const amount = receipt.events[0].args[0]
+    ethers.utils.formatEther(amount)
+    
+    // to compare with the value sent, we need to convert to tinybar
+    expect(amount.mul('10000000000')).to.equal(valueToSend)
+  })
+
+  it('tx.gasprice (uint): gas price of the transaction', async function () {
+    const gasPrice = await contractTr.getGasPrice()
+
+    expect(gasPrice).to.exist
+    expect(ethers.BigNumber.isBigNumber(gasPrice)).to.be.true
```

Comment:
Add greater than 0 check here and any other numbers.
0 gets used often as a default and can signal a bug

Reply:
Thanks, good to know.
I added `greater than 0` check. 
I am not sure I understand the second part of the comment `...and any other numbers.`

Reply:
Yeah i just meant any other values you may be checking in EVM responses you might want to confirm non zero also